pkg/parse/state.go

```diff
@@ -28,36 +28,40 @@ const (
 	maxRetryInterval             = time.Duration(5) * time.Minute
 )
 
-type sourceStatus struct {
+type commonStatus struct {
 	commit     string
 	errs       status.MultiError
 	lastUpdate metav1.Time
 }
 
-func (gs sourceStatus) equal(other sourceStatus) bool {
-	return gs.commit == other.commit && status.DeepEqual(gs.errs, other.errs)
+func (cs commonStatus) Equal(other commonStatus) bool {
+	return cs.commit == other.commit && status.DeepEqual(cs.errs, other.errs)
+}
+
+type sourceStatus struct {
+	commonStatus
+}
+
+func (gs sourceStatus) Equal(other sourceStatus) bool {
+	return gs.commonStatus.Equal(other.commonStatus)
 }
 
 type renderingStatus struct {
-	commit     string
-	message    string
-	errs       status.MultiError
-	lastUpdate metav1.Time
+	commonStatus
+	message string
 }
 
-func (rs renderingStatus) equal(other renderingStatus) bool {
-	return rs.commit == other.commit && rs.message == other.message && status.DeepEqual(rs.errs, other.errs)
+func (rs renderingStatus) Equal(other renderingStatus) bool {
```

Comment:
Why do we need to convert `equal` to `Equal`? This method is only called in the same pacakge.

Reply:
We don't _have_ to, but there's no reason to have private methods on a private struct. No one outside the package can access the struct methods anyway, unless they were on a public interface.

Reply:
The reason I asked this is that when I added the `equal` method back then, I was told that I should avoid having a public method on a private struct.

Reply:
That’s silly. It happens all the time. It’s common practice to make a private struct that satisfies a public interface and provide a constructor that returns the public interface as implemented by the private struct with public methods.

Reply:
The think you want to avoid is having a public function or method that returns a private struct in its signature, because the caller won’t be able to create a variable to store the return value.

Reply:
I see. Thanks for the explanation.